app/services/permissions_checker.rb

```diff
@@ -74,4 +79,20 @@ def authorize_room_limit
 
     true
   end
+
+  def current_provider_check
+    # check to see if current user is trying to access another provider
+    return false if @current_user.provider != @current_provider
+
+    # check to see if current_user is trying to access a user from another provider
+    return false if @user_id.present? && (@current_user.provider != User.find(@user_id.to_s).provider)
+
+    # check to see if current_user is trying to access a room from another provider
+    return false if @friendly_id.present? && (@current_user.provider != Room.find_by(friendly_id: @friendly_id.to_s).user.provider)
+
+    # check to see if current_user is trying to access a recording from another provider
+    return false if @record_id.present? && (@current_user.provider != Recording.find_by(record_id: @record_id.to_s).user.provider)
+
+    true
+  end
```

Comment:
I think it would be a bit easier to read if it was positive logic
`return true if ==`